### ISSUE TITLE
Non-resource group rule triggering for a resource group

### ISSUE THREAD

**Issue by anwather** [Azure/PSRule.Rules.Azure#2401]
Rule definition:
```
{
    "apiVersion": "github.com/microsoft/PSRule/v1",
    "kind": "Rule",
    "metadata": {
        "name": "Azure.Policy.654ac0e40740",
        "displayName": "Require a tag on resources",
        "tags": {
            "Azure.Policy/category": "Tags"
        },
        "annotations": {
            "Azure.Policy/id": "/providers/Microsoft.Authorization/policyDefinitions/871b6d14-10aa-478d-b590-94f262ecfa99",
            "Azure.Policy/version": "1.0.1"
        }
    },
    "spec": {
        "recommend": "Enforces existence of a tag. Does not apply to resource groups.",
        "with": [
            "PSRule.Rules.Azure\\Azure.Resource.SupportsTags"
        ],
        "condition": {
            "field": "tags['Environment']",
            "exists": true
        }
    }
}
```
This is triggering for a resource group bicep template - however if deployed into Azure it would pass as the policy doesn't support resource groups.
